packages/contractkit/src/wallets/rpc-wallet.test.ts

```diff
@@ -0,0 +1,241 @@
+  },
+  message: {
+    from: {
+      name: 'Cow',
+      wallet: '0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826',
+    },
+    to: {
+      name: 'Bob',
+      wallet: '0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB',
+    },
+    contents: 'Hello, Bob!',
+  },
+}
+
+export const PRIVATE_KEY1 = '1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef'
+export const ACCOUNT_ADDRESS1 = normalizeAddressWith0x(privateKeyToAddress(PRIVATE_KEY1))
+export const PRIVATE_KEY2 = '0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890fdeccc'
+export const ACCOUNT_ADDRESS2 = normalizeAddressWith0x(privateKeyToAddress(PRIVATE_KEY2))
+
+const PASSPHRASE = 'ce10'
+const DURATION = 100
+
+testWithGanache('rpc-wallet', (web3) => {
+  const provider = web3.currentProvider
+  const rpcWallet = new RpcWallet(provider)
+
+  describe('with ganache web3 provider', () => {
+    let web3Accounts: string[]
+
+    beforeAll(async () => {
+      web3Accounts = (await web3.eth.getAccounts()).map(normalizeAddressWith0x)
+      await rpcWallet.init()
+    })
+
+    test('initalizes with default accounts', async () => {
+      const accounts = rpcWallet.getAccounts()
+      expect(accounts).toEqual(web3Accounts)
+    })
+
+    test('fails if you add an invalid private key', async () => {
+      try {
+        await rpcWallet.addAccount('this is not a valid private key', PASSPHRASE)
+        throw new Error('Expected exception to be thrown')
+      } catch (e) {
+        expect(e.message).toBe('private key length is invalid')
+      }
+    })
+
+    test('succeeds if you add a private key without 0x', async () => {
+      await rpcWallet.addAccount(PRIVATE_KEY1, PASSPHRASE)
+      expect(rpcWallet.hasAccount(ACCOUNT_ADDRESS1)).toBeTruthy()
+    })
+
+    test('fails if you add a private key twice', async () => {
+      try {
+        await rpcWallet.addAccount(PRIVATE_KEY1, PASSPHRASE)
+        throw new Error('Expected exception to be thrown')
+      } catch (e) {
+        expect(e.message).toBe(`RpcWallet: account already exists`)
+      }
+    })
+
+    test('succeeds if you add a private key with 0x', async () => {
+      await rpcWallet.addAccount(PRIVATE_KEY2, PASSPHRASE)
+      expect(rpcWallet.hasAccount(ACCOUNT_ADDRESS2)).toBeTruthy()
+    })
+
+    describe('with added accounts', () => {
+      test('all addresses can be retrieved', () => {
+        expect(rpcWallet.getAccounts()).toEqual(
+          web3Accounts.concat([ACCOUNT_ADDRESS1, ACCOUNT_ADDRESS2])
+        )
+      })
+
+      describe('unlocking', () => {
+        test('fails if you use an invalid passphrase', async () => {
+          try {
+            await rpcWallet.unlockAccount(ACCOUNT_ADDRESS1, 'wrong_passphrase', DURATION)
+          } catch (e) {
+            expect(e.message).toContain('RpcSigner unlock failed')
+          }
+        })
+      })
+
+      // TODO(yorke): ENABLE WHEN GANACHE SUPPORTS THESE METHODS PROPERLY
+      // describe('signing', () => {
```

Comment:
You can use `xdescribe` or `describe.skip` to add tests that aren't yet passing.